homeassistant/components/schlage/switch.py

```diff
@@ -90,19 +90,23 @@ def __init__(
         self._attr_unique_id = f"{device_id}_{self.entity_description.key}"
 
     @property
-    def is_on(self) -> bool:
+    def is_on(self) -> bool | None:
         """Return True if the switch is on."""
+        if self._lock is None:
+            return None
```

Comment:
This looks like a workaround for availability?

If `self._lock` is `None`, shouldn't the whole entity be in an unavailable state?